optaplanner-core/src/main/java/org/optaplanner/core/impl/score/director/AbstractScoreDirector.java

```diff
@@ -378,8 +378,12 @@ public static String explainScore(Score workingScore,
     }
 
     @Override
-    public String explainScore() {
-        return explainScore(calculateScore(), getConstraintMatchTotalMap().values(), getIndictmentMap().values());
+    public <Score_ extends Score<Score_>> String explainScore() {
```

Comment:
Now wouldn't it be lovely if `AbstractScoreDirector` had a generic <Score_> type? :-) Inner and Abstract score directors, and their subclasses, are officially private code in 8, so adding more generics there does not affect users.

Reply:
Yes, I am starting to think so too. Make InnerScoreDirector (not ScoreDirector!) Score generic. Separate PR in any case. Want to create a jira for it?

Reply:
https://issues.redhat.com/browse/PLANNER-2164